src/Authoring/WinRT.SourceGenerator/Generator.cs

```diff
@@ -231,9 +232,23 @@ private bool HasSomePublicTypes(SyntaxNode syntaxNode)
 
         public void OnVisitSyntaxNode(SyntaxNode syntaxNode)
         {
-            // We will lose namespaces in `Declarations` if we don't special case them as exempt from the `IsPublic` check
-            if (syntaxNode is not MemberDeclarationSyntax decaralation || 
-                (syntaxNode is not NamespaceDeclarationSyntax @namespace && !IsPublic(decaralation)))
+            // Store namespaces separately as we only need to look at them for diagnostics
+            // If we did store them in declarations, we would get duplicate entries in the WinMD,
+            // once from the namespace declaration and once from the member's declaration
+            if (syntaxNode is NamespaceDeclarationSyntax @namespace)
+            {
+                if (!HasSomePublicTypes(syntaxNode))
+                {

```

Comment:
We might want to just return always if it enters this if namespace as it is not going to meet the checks after.